gqcp/include/Mathematical/Optimization/Eigenproblem/Eigenpair.hpp

```diff
@@ -19,56 +19,84 @@
 
 
 #include "Mathematical/Representation/Matrix.hpp"
+#include "Utilities/linalg.hpp"
 
 
 namespace GQCP {
 
 
 /**
- *  A container class to store an eigenpair, i.e. an eigenvector with a corresponding eigenvalue
+ *  A container class to store an eigenpair, i.e. an eigenvector with a corresponding eigenvalue.
  */
+template <typename _Scalar = double>
 class Eigenpair {
+public:
+    using Scalar = _Scalar;
+
 private:
-    double m_eigenvalue;
-    VectorX<double> m_eigenvector;
+    Scalar m_eigenvalue;
+    VectorX<Scalar> m_eigenvector;
 
 
 public:
     // CONSTRUCTORS
 
     /**
-     *  @param eigenvalue       the eigenvalue
-     *  @param eigenvector      the eigenvector
+     *  @param eigenvalue       The eigenvalue.
+     *  @param eigenvector      The eigenvector.
      */
-    Eigenpair(const double eigenvalue, const VectorX<double>& eigenvector);
+    Eigenpair(const Scalar eigenvalue, const VectorX<Scalar>& eigenvector) :
+        m_eigenvalue {eigenvalue},
+        m_eigenvector {eigenvector} {}
 
     /**
      *  A constructor that sets the eigenvalue to zero and the corresponding eigenvector to zeros
      *
-     *  @param dimension        the dimension of the eigenvector
+     *  @param dimension        The dimension of the eigenvector.
      */
-    explicit Eigenpair(const size_t dimension = 1);
+    explicit Eigenpair(const size_t dimension = 1) :
+        Eigenpair(0.0, VectorX<double>::Zero(dimension)) {}
 
 
     // PUBLIC METHODS
 
     /**
+     * A function that returns the eigenvalue associated to this eigenpair.
+     * 
      *  @return the eigenvalue associated to this eigenpair
      */
```

Comment:
```suggestion
    /**
     *  @return The eigenvalue associated to this eigenpair.
     */
```

I know that I mentioned to add extra explanation to member functions, but for trivial accessors, I don't think is required.

Also, "A function that ... " seems pretty redundant as documentation style.